cypress-io/cypress
# feat: create from React component

### User facing changelog


Users working in React projects can now generate a basic Component Testing spec file from the components that exist in their project.

Note: this was reviewed in several PRs - this PR does not introduce new code.

### Additional details


This feature will improve the onboarding experience for users creating Component Testing specs for the first time (or not for the first time) by reducing friction and allowing them to get a component mounted inside of a test in a matter of seconds  without reading any documentation or manually creating a new spec file.

Users will still be able to create empty specs from the UI as well.

### Steps to test


1. Open Component Testing with a project that uses React (CRA, Next, etc.)
2. Click the "New Spec" button in the top-right corner of the specs list
3. Select "Create from component" card
4. See a list of the files in your project that may contain React components
5. Expand each row and verify that any React components **that are exported** from that file are listed in the expanded list
6. Click one of the components and verify that a new spec is created that correctly imports and mounts the component you selected (the test may not pass if the component has required props or requires providers in the component tree)
7. Verify that the first time you run the spec there is a green banner with a link to the Styling Components documentation

### How has the user experience changed?


https://www.loom.com/share/c9ed4fafda0a4db49b0ac9fe047f4685

### PR Tasks


- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/4907
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
